R/teal_data_utils.R

```diff
@@ -0,0 +1,84 @@
+#' `teal_data` utils
+#'
+#' In `teal` we need to recreate the `teal_data` object due to two operations:integer.max
+#' - we need to append filter-data code and objects which have been evaluated in `FilteredData` and
+#' we want to avoid double-evaluation.
+#' - we need to subset `teal_data` to datanames used by the module, to shorten obtainable R-code
+#'
+#' Due to above recreation of `teal_data` object can't be done simply by using public
+#' `teal.code` and `teal.data` methods.
+#'
+#' @param data (`teal_data`)
+#' @param code (`character`) code to append to `data@code`
+#' @param objects (`list`) objects to append to `data@env`
+#' @param datanames (`character`) names of the datasets
+#' @return modified `teal_data`
+#' @name teal_data_utilities
+NULL
+
+#' @rdname teal_data_utilities
+.append_evaluated_code <- function(data, code) {
+  checkmate::assert_class(data, "teal_data")
+  data@code <- c(data@code, code)
+  data@id <- c(data@id, sample.int(.Machine$integer.max, size = length(code)))
```

Comment:
I know it's very uncommon but what if `sample.int(.Machine$integer.max, size = length(code))` creates an `id` that already existed in `data@id`? I think below suggestion isn't great as we create `1:.Machine$integer.max` vector each time it is using this function.
```suggestion
  data@id <- c(data@id, sample(setdiff(1:.Machine$integer.max, data@id), size = length(code)))
```


```r
> object.size(1:1:.Machine$integer.max)
8589934640 bytes
> object.size(1:1:.Machine$integer.max)/1000000
8589.9 Mbytes
```


Maybe instead of sampling ids, we can just increment them, by always taking the highest id?

```r
data@id <- c(data@id, sample((max(data@id)+1):(max(data@id)+length(data@id)))
```

Reply:
or

```r
data@id <- c(data@id, max(data@id)+sample(1:length(data@id)))
```

Reply:
We talk about 2147483647 possible values. 
Probability that the sampled int will be duplicated is almost the same as the probability that `max(data@id)` will be close to `.Machine$integer.max` (which will also cause some failure).

`1:.Machine$integer.max, data@id)`

Reply:
What about the extension by increment? Taking max and then extending by the lenght? 
sample always take some time for sampling. Check below

```r


library(microbenchmark)

existing_ids <- sample(1:100, size = 50)

extend_by_max <- function(ids) {
  c(ids, (max(ids)+1):length(ids))
}
extend_by_rdm <- function(ids){
  c(ids, sample(.Machine$integer.max, size = length(ids)))
}

microbenchmark(
  extend_by_max(existing_ids), 
  extend_by_rdm(existing_ids)
)
```

```r
Unit: nanoseconds
                        expr  min   lq  mean median   uq     max neval cld
 extend_by_max(existing_ids)  700  900 19337    950 1100 1819900   100   a
 extend_by_rdm(existing_ids) 8400 8700 20726   9000 9200 1160400   100   a
```

Reply:
I changed sample.int to seq 👍

Reply:
@m7pr please wait until i fix the checks

Reply:
@gogonzo should we apply the same `seq` strategy in the constructor 

https://github.com/insightsengineering/teal.data/blob/b8f4522142075331f98b461b7a16f01edea88403/R/teal_data-class.R#L80

I think you can push directly to `669_insertUI@main` branch on `teal.data` repository